app/services/irs_impact_analysis_csv_service.rb

```diff
@@ -0,0 +1,103 @@
+require 'csv'
+
+class IrsImpactAnalysisCsvService
+  CSV_FIELDS = %w{
+    legal_name
+    last_four_ssn
+    date_of_birth
+    zip_code
+    reached_gathering_docs
+    uploaded_documents
+    completed_intake
+    filed
+    service_type
+  }.freeze
+
+  CSV_HEADERS = CSV_FIELDS.map { |field| field.gsub(/\W/, "") }.freeze
+
+  # only grab intakes where the user consented.
+  def intakes
+    Intake.where.not(primary_consented_to_service_at: nil).includes(:ticket_statuses, :documents)
+  end
+
+  def generate_csv
+    CSV.generate(headers: csv_headers, write_headers: true) do |csv|
+      intakes.find_each(batch_size: 100) do |intake|
+        csv << csv_row(decorated_intake(intake))
+      end
+    end
+  end
+
+  def send_csv
+    # TODO: Whats the best way to download this document?
+  end
+
+  def csv
+    @csv ||= generate_csv
+  end
+
+  def decorated_intake(intake)
+    IRSImpactIntake.new(intake)
+  end
+
+  private
+
+  def csv_headers
+    CSV_HEADERS
+  end
+
+  def csv_row(intake)
+    row = CSV_FIELDS.map { |field| intake.send(field) }
+  end
+
+  class IRSImpactIntake < SimpleDelegator
+    def legal_name
+      "#{primary_first_name} #{primary_last_name}"
+    end
+
+    def last_four_ssn
+      primary_last_four_ssn
+    end
+
+    def date_of_birth
+      primary_birth_date.strftime("%m/%d/%Y")
+    end
+
+    def reached_gathering_docs
+      key = EitcZendeskInstance::INTAKE_STATUS_GATHERING_DOCUMENTS
+      verified_ticket_statuses.detect { |ts| ts.intake_status == key }.present?
+    end
```

Comment:
I think this is the right approach. But reading it, I realize that we should give the caveat that this means the ticket _reached this status at some point_ and so theoretically could include tickets that were accidentally given this status and then reverted to a prior status.

Reply:
does the verified: true cover the transition back and forth?

Reply:
no, verified true tells us if we know the _time_ that it reached that status. verified false means that we know it was in that status at some point, but we don't know when it _reached_ the status.

Reply:
okay, ill let kelly + Day know about the caveat!